runtime/tests/checker/entitlements_test.go

```diff
@@ -5515,4 +5515,44 @@ func TestCheckIdentityMapping(t *testing.T) {
 
 		assert.NoError(t, err)
 	})
+
+	t.Run("owned value, with entitlements, function typed field", func(t *testing.T) {
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            entitlement A
+            entitlement B
+            entitlement C
+
+            struct X {
+               access(A | B) var s: String
+
+               init() {
+                   self.s = "hello"
+               }
+
+               access(C) fun foo() {}
+            }
+
+            struct Y {
+
+                access(Identity) let fn: (fun (): X)
+
+                init() {
+                    self.fn = fun(): X {
+                        return X()
+                    }
+                }
+            }
+
+            fun main() {
+                let y = Y()
+                let v = y.fn()
+            }
+        `)
+
+		errors := RequireCheckerErrors(t, err, 1)
+		invalidMapping := &sema.InvalidMappedEntitlementMemberError{}
```

Comment:
Oh hmm, if functions are just disallowed here maybe use a reference to a function type so we can actually test the mapping behavior.

Reply:
added in 842a9cd482a53e47943756ca6817c45774c531ea